### DISCUSSION TITLE
Color settings don't work

### DISCUSSION THREAD

**Discussion by red0888** [retypeapp/retype#127] [Q&A] [answered]
Hello, I tried changing the primary color for my retype website. Even after setting the colors as shown here- https://retype.com/configuration/project/#colors I had no luck, everything is still in blue.

**Comment by geoffreymcgill** [comment, marked as answer]
Hi. Currently, only customizing the color of the label is supported. 

<img width="307" alt="Screen Shot 2021-08-18 at 3 55 37 AM" src="https://user-images.githubusercontent.com/62210/129878878-581f3ddd-3548-433f-96c2-dfc2f34e84aa.png">

We are working on new functionality to customize the entire palette of the website, but there's still work to be done and I'm not exactly sure when it will be released, although the palette functionality will likely be included somewhere within the next three releases.

**Reply by red0888** [reply]
Thank you for the heads up :D, it would be nice if colors can be set entirely in the YML file, and if there are just 3-4 shades of the color. I personally liked how docusaurus provided a tool to easily get shades of a color with the help of 1 primary color. https://docusaurus.io/docs/styling-layout#styling-your-site-with-infima

**Reply by DannyBen** [reply]
Just to chime in here with a contrarian opinion:

From my experience with working with lightweight website builder and documentation builders, once you start adding color configuration in the configuration file, it eventually turn into a feature creep of configuration options that are related to styling, which end up usually just disappointing the next group of users, who wanted to further customize something that is not available.

Then you either end up with endless requests for "why can't I customize this element" and end up with either a large, usually awkward, styling configuration, or end up with creating a separate "theme" configuration, or end up allowing custom CSS overrides.

I guess what I am trying to say is this: Before diving head first into enabling *some* customization in the generic configuration file, I would invest more time in considering what is the vision of Retype in this regard, and build it in a way that fits that vision from the start (i.e., is Retype "opinionated", or does it want to let you change many things freely).

**Comment by red0888** [comment]
Good idea. If you want to keep retype opionionated, then it would be a good idea to consider adding color themes to pick from, where the user can just specify the theme in the YML file. Perhaps, further advanced component customization could be a part of Pro/Enterprise plan if and when they become available.
